### PR TITLE
keto: init at 0.13.0-alpha.0

### PR DESCRIPTION
**Pull request by MrMebelMan**
## Description of changes
Add a new package

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ke/keto/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ fetchFromGitHub, buildGoModule, lib, stdenv }:
```
Comment (DontEatOreo) [inline pkgs/by-name/ke/keto/default.nix:1]:
```suggestion
{
  fetchFromGitHub,
  buildGoModule,
  lib,
}:
```

File: pkgs/by-name/ke/keto/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+buildGoModule rec {
+  pname = "keto";
+  version = "0.13.0-alpha.0";
```
Comment (DontEatOreo) [inline pkgs/by-name/ke/keto/default.nix:5]:
```suggestion
let
  pname = "keto";
  version = "0.13.0-alpha.0";
in
buildGoModule {
  inherit pname version;
```
Using `rec` is discouraged, please take a look at: https://nix.dev/guides/best-practices#recursive-attribute-set-rec

File: pkgs/by-name/ke/keto/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "-w"
+    "-X github.com/ory/keto/internal/driver/config.Version=${version}"
+    "-X github.com/ory/keto/internal/driver/config.Commit=c75695837f170334b526359f28967aa33d61bce6"
+  ];
+
+  meta = with lib; {
+    description = "ORY Keto, the open source access control server";
+    homepage = "https://www.ory.sh/keto/";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ mrmebelman ];
```
Comment (DontEatOreo) [inline pkgs/by-name/ke/keto/default.nix:30]:
```suggestion
  meta = {
    description = "ORY Keto, the open source access control server";
    homepage = "https://www.ory.sh/keto/";
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ mrmebelman ];
```

using `with` is discouraged, please take a look at https://nix.dev/guides/best-practices#with-scopes and

File: pkgs/by-name/ke/keto/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  src = fetchFromGitHub {
+    owner = "ory";
+    repo = "keto";
+    rev = "v${version}";
+    sha256 = "sha256-0yylaaXogN2HWXY8Tb7ScN4jdyeHecJ0gBYlVvcwaNE=";
+  };
+
+  vendorHash = "sha256-lgwV4Ysjmd9e850Rf5c0wSZtMW3U34/piwwG7dQEUV4=";
+
+  tags = [ "sqlite" "json1" "hsm" ];
```
Comment (DontEatOreo) [inline pkgs/by-name/ke/keto/default.nix:16]:
```suggestion
  tags = [
    "sqlite"
    "json1"
    "hsm"
  ];
```

**Review by 4evy — CHANGES_REQUESTED** [review]

**Comment by DontEatOreo** [comment]
`pkgs/by-name` automatically detects the package no need to add it to `pkgs/top-level/all-packages.nix`

**Comment by MrMebelMan** [comment]
> `pkgs/by-name` automatically detects the package no need to add it to `pkgs/top-level/all-packages.nix`

When I remove it from all-packages, I'm getting an error: 
image

**Comment by DontEatOreo** [comment]
You need to pass down the package name in the attr, so the command will be `nix-build -A keto`

**Comment by DontEatOreo** [comment]
The new file in `pkgs/by-name` should be named `package.nix` instead of `default.nix`

**Comment by MrMebelMan** [comment]
@DontEatOreo when I rename it from default.nix to package.nix, the build crashes with

`error: getting status of '/home/vlad/nixpkgs/pkgs/by-name/ke/keto/default.nix': No such file or directory`

**Comment by MrMebelMan** [comment]
@DontEatOreo thanks!

**Comment by prusnak** [comment]
@GrahamcOfBorg build keto

**Pull request merged**